heudiconv/tests/test_regression.py

```diff
@@ -81,3 +82,47 @@ def test_multiecho(tmpdir, subject='MEEPI', heuristic='bids_ME.py'):
     events = glob(op.join('out', 'sub-' + subject, 'func', '*events.tsv'))
     for event in events:
         assert 'echo-' not in event
+
+
+@pytest.mark.parametrize('subject', ['merged'])
+def test_grouping(tmpdir, subject):
+    dicoms = [
+        op.join(TESTS_DATA_PATH, fl) for fl in ['axasc35.dcm', 'phantom.dcm']
+    ]
+    # ensure DICOMs are different studies
+    studyuids = {
+        dcm.read_file(fl, stop_before_pixels=True).StudyInstanceUID for fl
+        in dicoms
+    }
+    assert len(studyuids) == len(dicoms)
+    # symlink to common location
+    outdir = tmpdir.mkdir('out')
+    datadir = tmpdir.mkdir(subject)
+    for fl in dicoms:
+        os.symlink(fl, (datadir / op.basename(fl)).strpath)
+
+    template = op.join("{subject}/*.dcm")
+    hargs = gen_heudiconv_args(
+        tmpdir.strpath, outdir.strpath, subject, 'convertall.py',
+        template=template
+    )
+
+    with pytest.raises(AssertionError):
+        """
+    test_regression.py:110: in test_grouping
+    runner(hargs)
+../cli/run.py:128: in main
+    process_args(args)
+../cli/run.py:333: in process_args
+    dcmconfig=args.dcmconfig,)
+../convert.py:168: in prep_conversion
+    grouping=None)
+../dicoms.py:96: in group_dicoms_into_seqinfos
+    studyUID, file_studyUID
+E   AssertionError: Conflicting study identifiers found [1.3.12.2.1107.5.2.32.35131.30000014022817282751500000052, 1.3.12.2.1107.5.2.43.66112.30000016081515591223100000010].
+"""
+        runner(hargs)
+
+    # group all found DICOMs under subject, despite conflicts
+    hargs += ["-g", "all"]
+    runner(hargs)
```

Comment:
may be things are just skipped?  could we check the output data to be all converted as expected?

Reply:
this comment was not addressed yet. Let's make tests stronger -- this is a big and important change, so better be tested thoroughly

Reply:
i addressed this in my latest commit 👍